CMakeLists: enable OpenEmbedded compilation

amstewart opened the pull request:
### What does this Pull Request accomplish?

This patchset makes several changes to the CMakeLists file which enable cross-compilation in the context of an OpenEmbedded build environment. In the OE-case, it is not reasonable to satisfy build dependencies using git dependencies.

### Why should this Pull Request be merged?

This patchset is needed to enable the `ni-grpc-device` recipe in the [meta-nilrt:nilrt/master/dunfell](https://github.com/ni/meta-nilrt/tree/nilrt/master%2Fdunfell) repo.

### What testing has been done?

OE builds this CMakeList successfully on my dev machine. I expect the GH workflows can verify that it builds correctly in the mundane windows/linux build context.

SparkingSpork commented on CMakeLists.txt:
```diff
@@ -208,12 +218,17 @@ add_custom_command(
+endif()
 
 # Add googletest tests
 enable_testing()
 set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
```
Per the Teams conversation, move these so that they're called prior to adding gtest. I think it should be OK for them to be outside of the if block.

amstewart replied:
I think I like the idea that testing won't be enabled at all in the NILRT-OE case. I'll see if I can get it working within the branch structure.

SparkingSpork commented on CMakeLists.txt:
```diff
@@ -208,12 +218,17 @@ add_custom_command(
 #----------------------------------------------------------------------
 # Add JSON parser
 #----------------------------------------------------------------------
```
We should probably update this comment to indicate gtest is added here now too.

amstewart replied:
Will fix.

amstewart commented:
## Update 2
* Added a new cmake variable `_GRPC_DEVICE_NILRT_LEGACY_TOOLCHAIN`. When `TRUE`, it asserts that CMakeLists should setup the toolchain dependencies to satisfy the published NIRLT toolchain archive. This change fixes the "NILRT" GH workflow.
* Fixed the Windows compilation workflow by moving the `set(gtest_force_shared_crt...` line before `add_subdirectory(third_party/gtest...`. Thanks; @SparkingSpork.

All workflows should pass now.

Ready for final review and merge.
